outlines/models/openai.py

```diff
@@ -1,297 +1,156 @@
+    def format_str_input(self, model_input: str):
+        """Generate the `messages` argument to pass to the client when the user
+        only passes a prompt.
 
-    def stream(self, *args, **kwargs):
+        """
+        return {
+            "messages": [
+                {
+                    "role": "user",
+                    "content": model_input,
+                }
+            ]
+        }
+
+    @format_input.register(Vision)
+    def format_vision_input(self, model_input: Vision):
+        """Generate the `messages` argument to pass to the client when the user
+        passes a prompt and an image.
+
+        """
+        return {
+            "messages": [
+                {
+                    "role": "user",
+                    "content": [
+                        {"type": "text", "text": model_input.prompt},
+                        {
+                            "type": "image_url",
+                            "image_url": {
+                                "url": f"data:{model_input.image_format};base64,{model_input.image_str}"
+                            },
+                        },
+                    ],
+                }
+            ]
+        }
+
+    @singledispatchmethod
+    def format_output_type(self, output_type):
+        """Generate the `response_format` argument to the client based on the
+        output type specified by the user.
+
+        """
         raise NotImplementedError(
-            "Streaming is currently not supported for the OpenAI API"
+            f"The type {output_type} is not available with OpenAI. The only output type available is `Json`."
         )
 
-    def new_with_replacements(self, **kwargs):
-        new_instance = copy.copy(self)
-        new_instance.config = replace(new_instance.config, **kwargs)
-        return new_instance
-
-    def __str__(self):
-        return self.__class__.__name__ + " API"
-
-    def __repr__(self):
-        return str(self.config)
-
-
-@functools.partial(vectorize, signature="(),(),(),()->(s),(),()")
-async def generate_chat(
-    prompt: str,
-    system_prompt: Union[str, None],
-    client,
-    config: OpenAIConfig,
-) -> Tuple[np.ndarray, int, int]:
-    """Call OpenAI's Chat Completion API.
-
-    Parameters
-    ----------
-    prompt
-        The prompt we use to start the generation. Passed to the model
-        with the "user" role.
-    system_prompt
-        The system prompt, passed to the model with the "system" role
-        before the prompt.
-    client
-        The API client
-    config
-        An `OpenAIConfig` instance.
-
-    Returns
-    -------
-    A tuple that contains the model's response(s) and usage statistics.
+    @format_output_type.register(NoneType)
+    def format_none_output_type(self, _: None):
+        """Generate the `response_format` argument to the client when no
+        output type is specified by the user.
 
-    """
+        """
+        return {}
 
-    @error_handler
-    @cache()
```

Comment:
I see `@cache()` is out, is there any plan for it?

Reply:
Good question. I think we need to re-think caching across the library. I am ok with dropping it for now. Wdyt?

Reply:
Yeah, I agree 👍